From 74b0cb5b86f7d7f8f7c1172d85b09096bef147b7 Mon Sep 17 00:00:00 2001
From: Daniel Nicoara <dnicoara@chromium.org>
Date: Thu, 24 Sep 2020 02:34:24 +0000
Subject: [PATCH] Remove dead-reloc-in-nonalloc LD flags

Breakpad change landed. Revert workaround.

Bug: 1105559
Test: components/crash/content/tools/generate_breakpad_symbols.py --build-dir=out/andrd --binary=out/andrd/lib.unstripped/libcontent_shell_content_view.so --symbols-dir=/tmp/foo --platform=android
Change-Id: I519534002898a97b15a57b9b87ac78ef3f216dee
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2427349
Commit-Queue: Nico Weber <thakis@chromium.org>
Reviewed-by: Nico Weber <thakis@chromium.org>
Cr-Commit-Position: refs/heads/master@{#810066}
---
 build/config/compiler/BUILD.gn | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index aa9eca20bd2..42839cfba3f 100644
--- build/config/compiler/BUILD.gn
+++ build/config/compiler/BUILD.gn
@@ -396,14 +396,6 @@ config("compiler") {
         "-Wl,--as-needed",
       ]
     }
-    if (use_lld && !is_chromeos_device) {
-      # TODO(thakis): Fix dump_syms to not need this and then remove it,
-      # https://crbug.com/1105559
-      ldflags += [
-        "-Wl,-z,dead-reloc-in-nonalloc=*=0",
-        "-Wl,-z,dead-reloc-in-nonalloc=.debug_ranges=1",
-      ]
-    }
   }
 
   # Linux-specific compiler flags setup.
